doc/upgrading.md

```diff
@@ -2,21 +2,29 @@
 This guide describes how to upgrade your SPIRE deployment, as well as the compatibility guarantees that SPIRE users can expect.
 
 ## SPIRE Versioning
-SPIRE versions are expressed as **x.y.z**, where **x** is the major version, **y** is the minor version, and **z** is the patch version, following Semantic Versioning terminology. Despite the fact that the project is currently pre-1.0, version compatibility guarantees are already in place. This document will be updated in advance of a 1.0.0 release.
+SPIRE versions are expressed as **x.y.z**, where **x** is the major version, **y** is the minor version, and **z** is the patch version, following Semantic Versioning terminology. The last pre-1.0 versions are 0.12.x, which as an exception have compatibility warranties with 1.0.x. Versions prior to 0.12.0 are not compatible with 1.0.x.
 
 ### SPIRE Server Compatibility
-Version skew within a SPIRE Server cluster is supported within +/- 1 minor version. In other words, the newest and oldest SPIRE Server instances in any given cluster must be within one minor version of each other.
+Version skew within a SPIRE Server cluster is supported within +/- 1 minor version. In other words, the newest and oldest SPIRE Server instances in any given cluster must be within one minor version of each other. As an exception, versions 0.12.x are compatible with 1.0.x versions.
 
-For example:
-* Newest SPIRE Server instance is at 0.9.3
-* Other SPIRE Server instances are supported at 0.9.x and 0.8.x
+Example 1 (0.12.x exception):
+* Newest SPIRE Server instance is at 1.0.3
+* Other SPIRE Server instances are supported at 1.0.x and 0.12.x
+
+Example 2:
+* Newest SPIRE Server instance is at 1.2.3
+* Other SPIRE Server instances are supported at 1.2.x and 1.1.x
 
 ### SPIRE Agent Compatibility
-SPIRE Agents must not be newer than the oldest SPIRE Server that they communicate with, and may be up to one minor version older.
+SPIRE Agents must not be newer than the oldest SPIRE Server that they communicate with, and may be up to one minor version older. As an exception, versions 0.12.x are compatible with 1.0.x versions (as long as the oldest SPIRE Server is at 1.0.x).
```

Comment:
I'm finding this edit a little confusing. I think it would be clearer if we stated "agent versions ... are compatible with server versions ...".

Is the comment in parenthesis at the end strictly true? I'm having trouble imagining the scenario

Reply:
Now that I think it better, it doesn't make a lot of sense to add that comment in parenthesis.
I'll rephrase the edit removing it.